frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/AttachmentStackedBar.tsx

```diff
@@ -28,65 +31,99 @@ const FilledTrack = (props: BoxProps) => {
 }
 
 export const AttachmentStackedBar = ({
+  existingValues: existingValuesProp,
+  newValue = 0,
   min = 0,
   max,
-  values,
 }: AttachmentStackedBarProps): JSX.Element => {
+  const isLoading = useMemo(() => !existingValuesProp, [existingValuesProp])
+
+  const existingValues = useMemo(
+    () => existingValuesProp ?? [],
+    [existingValuesProp],
+  )
+
+  const totalAttachmentSize = useMemo(
+    () => existingValues.reduce((sum, v) => sum + v, 0) + newValue,
+    [existingValues, newValue],
+  )
+
   const isOverQuota = useMemo(
-    () => !!values && max < values.reduce((sum, v) => sum + v, 0),
-    [max, values],
+    () => !!existingValues && max < totalAttachmentSize,
```

Comment:
`!!existingValues` will always be true, since it is an array. this check may not be needed, or `existingValuesProp` used, or checking with `existingValues.length` instead?

Reply:
hmm, you're right, this check probably not needed (since anyway we set the skeleton into a loading state when `existingValuesProp` is not provided). will get rid of it!

Reply:
removed in [270f849](https://github.com/opengovsg/FormSG/pull/4565/commits/270f849d96cc309b531c5688e1f0464de2f3ab91)